pkg/daemon/cleanup/cleanup.go

```diff
@@ -57,13 +58,22 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 		klog.Exitf("could not create service manager: %s", err.Error())
 	}
 	defer svcMgr.Disconnect()
-	services, err := getServicesToRemove(ctx, directClient, node, configMapNamespace)
+	// get data from the latest services ConfigMap
```

Comment:
You are getting the latest service ConfigMap, passing it to `getServicesToRemove` and `ensureEnvVarsAreRemoved` and within each function getting the version ConfigMap and falling back to the latest if that fails. Shouldn't figuring out which ConfigMap to use be done outside of the two functions? This was we also ensure that both methods are using the same ConfigMap.

Reply:
updated, the logic was a bit complex for removing the services, I pulled this out.

Reply:
My comment was more of a question rather than a request for a change. In future please respond if it is making the code more complex before making the changes.